Mastodon/Extension/AppContext+NextAccount.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  AppContext+NextAccount.swift
+//  Mastodon
+//
+//  Created by Marcus Kida on 17.11.22.
+//
+
+import CoreData
+import CoreDataStack
+import MastodonCore
+import MastodonSDK
+
+extension AppContext {
+    func nextAccount(in authContext: AuthContext) -> MastodonAuthentication? {
+        let request = MastodonAuthentication.sortedFetchRequest
+        guard
+            let accounts = try? managedObjectContext.fetch(request),
+            accounts.count > 1
+        else { return nil }
+        
+        let nextSelectedAccountIndex: Int? = {
+            for (index, account) in accounts.enumerated() {
+                guard account == authContext.mastodonAuthenticationBox
+                    .authenticationRecord
+                    .object(in: managedObjectContext)
```

Comment:
Here we could use the `objectID` for `account` to compare the same object without database access.